src/components/Content.tsx

```diff
@@ -47,6 +55,66 @@ const Content: FC = () => {
     return children.length
   })
   const isAbsoluteContext = useSelector(state => isAbsolute(state.rootContext))
+  const visibilityRef = useRef<'above' | 'below' | null>(null)
```

Comment:
I did not use state because it would cause too many renders during drag but if it can be calculated with existing state maybe I need to think about it again.

Reply:
`useRef` and `useMemo` can both be used to persiste state across render, but for different reasons.  

`useRef` is used when you need _different_ information from the last render (e.g. the previous cursor location). 

`useMemo` is used when you need the _same_ information from the last render (for performance optimization).